_posts/2021-02-26-feather-nrf52840.md

```diff
@@ -0,0 +1,457 @@
+---
+layout: blog
+title: How to Port a Board for RIOT
+authors: Martine S. Lenders
+githubuid: miri64
+date: 2021-02-26
+tags: riot, board, microcontroller, adafruit, feather, nrf52
+description:
+    Our contributor Martine Lenders documents her experience of porting her very
+    first board for RIOT
+image: feather-nrf52840.jpg
+---
+
+*This was originally published on [Martine's blog in February
+2020][original-post].*
+
+# Introduction
+
+It is already a few weeks back, but this year marked a debut for me:
+Though I am a long-term contributor to RIOT—my oldest
+contribution [goes back to June 2011][a2c7cf8]—I never actually ported a board
+for RIOT support. This [changed at the beginning of this year][PR].
+
+RIOT is an operating system for the Internet of Things (IoT), so an operating
+system for hardware platforms that do not typical PC operating systems fit like
+Linux, Mac OSX, or Windows or even phone operating systems like Android or iOS.
+When talking of IoT devices, the RIOT community usually thinks in terms of
+[kilobytes of memory][RFC 7228]. Not gigabytes, not megabytes, _kilobytes_! As
+in, a few 1000 _bytes_. The premise behind this thinking is, that IoT devices
+can not follow [Moore's Law]—which states that every two years the number of
+transistors in a given piece of computer hardware approximately doubles. In
+fact, the efforts to save cost and energy for IoT hardware are directly opposed
+to Moore's Law: every additional transistor, be it for calculation or storage,
+costs more money and leads to more energy usage. Given, that for the IoT a
```

Comment:
Is that the case? My interpretation of Moore's Law was always, that this is exactly not the case. Otherwise, one could argue that Moore's Law would still hold true if would have vacuum tubes in skyscraper sized computers.